packages/job-worker/src/blueprints/context/adlibActions.ts

```diff
@@ -61,15 +62,65 @@ import { isPartPlayable } from '@sofie-automation/corelib/dist/dataModel/Part'
 import { moveNextPartInner } from '../../playout/playout'
 import _ = require('underscore')
 import { ProcessedShowStyleConfig } from '../config'
+import { DatastorePersistenceMode } from '@sofie-automation/shared-lib/dist/core/model/TimelineDatastore'
 
 export enum ActionPartChange {
 	NONE = 0,
 	SAFE_CHANGE = 1,
 }
 
+export class DatastoreActionExecutionContext
+	extends ShowStyleUserContext
+	implements IDataStoreActionExecutionContext, IEventContext
+{
+	protected readonly _context: JobContext
+
+	constructor(
+		contextInfo: UserContextInfo,
+		context: JobContext,
+		showStyle: ReadonlyDeep<ShowStyleCompound>,
+		watchedPackages: WatchedPackagesHelper
+	) {
+		super(contextInfo, context, showStyle, watchedPackages)
+		this._context = context
+	}
+
+	async setTimelineDatastoreValue(key: string, value: unknown, mode: DatastorePersistenceMode): Promise<void> {
+		const studioId = this._context.studioId
+		const id = protectString(`${studioId}_${key}`)
+		const collection = this._context.directCollections.TimelineDatastores
+
+		await collection.replace({
+			_id: id,
+			studioId: studioId,
+
+			key,
+			value,
+
+			modified: Date.now(),
+			mode,
+		})
+	}
+
+	async removeTimelineDatastoreValue(key: string): Promise<void> {
+		const studioId = this._context.studioId
+		const id = protectString(`${studioId}_${key}`)
```

Comment:
It would be good to define this id generation as a function, to ensure it is kept consistent.